pkg/operator/operator.go

```diff
@@ -241,76 +262,229 @@ func (k *Kontroller) withLeaderElection() error {
 func (k *Kontroller) process() {
 	glog.V(4).Info("Going through a loop cycle")
 
-	nodelist, err := k.nc.List(v1meta.ListOptions{})
+	// find nodes with the reboot-action=post label and check if all actions are complete.
+	// if all actions are completed successfully then remove the reboot-action=post label
+	// and set reboot-ok=false, telling the agent that the reboot has completed.
+	err := k.checkPostRebootActions()
 	if err != nil {
-		glog.Infof("Failed listing nodes %v", err)
+		glog.Errorf("Failed to check post-reboot actions: %v", err)
 		return
 	}
 
-	glog.V(6).Infof("Found nodes: %+v", nodelist.Items)
+	// find nodes which just rebooted but aren't yet running post-reboot actions.
+	// remove any old action annotations and add the reboot-action=post label.
+	// this starts the post-reboot actions.
+	err = k.markPostRebootActionable()
+	if err != nil {
+		glog.Errorf("Failed to update recently rebooted nodes: %v", err)
+		return
+	}
 
-	justRebootedNodes := k8sutil.FilterNodesByAnnotation(nodelist.Items, justRebootedSelector)
+	// find nodes with reboot-action=pre label and check if all actions are complete.
+	// if all actions are completed successfully then remove the reboot-action=pre label
+	// and set reboot-ok=true, telling the agent it's time to reboot.
+	err = k.checkPreRebootActions()
+	if err != nil {
+		glog.Errorf("Failed to check pre-reboot actions: %v", err)
+		return
+	}
 
-	if len(justRebootedNodes) > 0 {
-		glog.Infof("Found %d rebooted nodes, setting annotation %q to false", len(justRebootedNodes), constants.AnnotationOkToReboot)
+	// take some number of the rebootable nodes.
+	// remove any old action annotations and add the reboot-action=pre label.
+	// this starts the pre-reboot actions.
+	err = k.markPreRebootActionable()
```

Comment:
I can see how method names like these developed, but looking at it with fresh eyes, it seems nonsensical. If we went with the nomenclature I mentioned, this becomes `markBeforeReboot()` which adds a before-reboot=true annotation.